### PR TITLE
TypeScript support

### PR DESCRIPTION
**Pull request by akwanpdf**
Migrated JS source files to TS files to provide TS support.

We have provided branches to experiment on:
1. `[Your name]/ts-test` - branch that you write your code on. After coding, running `npm start` in the outer repo will auto-generate JS files from your edited TS files. These JS files will be gitignored when making PR.
2. `[Your name]/ts-dev` - branch that you merge the PR into. Once merged, changes will be automatically applied to branch 3.
3. `[Your name]/ts-master` - branch which "customer" would use as a dependency (for example, you would add this as a dependency to https://github.com/PDFTron/react-native-sample)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -202,7 +204,7 @@ import { DocumentView, RNPdftron } from 'react-native-pdftron';
 type Props = {};
 export default class App extends Component<Props> {
 
-  constructor(props) {
+  constructor(props) { // may need to use `props: Props` in TypeScript
```
Comment (sgong-pdftron) [inline README.md:207]: not sure what this sentence means, you can show 2 varients to be clear

File: .github/workflows/jsBuilder.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -m "Updating JS files" -a 
+
+      - name: Push changes
+        uses: ad-m/github-push-action@v0.6.0
+        with:
+          force: true
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ts-master # <-- Change this to the branch where you want builds to be (for example, "master" if you want changes to the primary branch to be applied to `master`).
```
Comment (sgong-pdftron) [inline .github/workflows/jsBuilder.yml:42]: so basically this needs to be changed to master prior to merge?

Comment (akwanpdf) [reply .github/workflows/jsBuilder.yml:42]: yes, this is changed to `master`

File: API.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # PDFTron React Native API
 
+## TypeScript
+
+PDFTron React Native now has support for TypeScript. Since not all customers use the language, the typings used in this document will be described using normal JavaScript types. For TypeScript users, type information is automatically provided while coding. They can also be found in our [TypeScript source files](src).
```
Comment (sgong-pdftron) [inline API.md:5]: not sure what's the significant of this paragraph here? what do you expect users to do after landing on a bunch of source code?

File: README.md [hunk 4]
```diff
@@ -321,6 +325,35 @@ const styles = StyleSheet.create({
...
 
+## TypeScript
+
+PDFTron React Native introduced support for TypeScript in version 3.0.0. This update mainly benefits those who already use TypeScript in their applications. It also provides certain benefits to all customers, including those who use JavaScript without TypeScript.
+
+To get access to this update, simply update your PDFTron React Native dependency to version 3.0.0 or higher.
+
+### All Users
+
+For non-TypeScript users, this update will not automatically install TypeScript, and they can continue to use the library as before.
```
Comment (sgong-pdftron) [inline README.md:336]: not sure what is "this update"

File: README.md [hunk 5]
```diff
@@ -321,6 +325,35 @@ const styles = StyleSheet.create({
...
+- Updated Add an API guides for [Android](https://www.pdftron.com/documentation/android/get-started/react-native/add-an-api/) and [iOS](https://www.pdftron.com/documentation/ios/get-started/react-native/add-an-api) to offer step-by-step, TypeScript-supported examples on accessing properties, methods, and events. Contributors should be aware that PRs will now be made to the `dev` branch (see [Contributing](./CONTRIBUTING.md)).
+
+If you have questions, head to our FAQ's Integration section for [Android](https://www.pdftron.com/documentation/android/faq) and [iOS](https://www.pdftron.com/documentation/ios/faq/).
+
+### TypeScript Users
+
+If you are an existing TypeScript user, then our custom typings will be available to you simply by updating your dependencies.
+
+Note:
+- We recommend that our library is used with at least TypeScript v3.4.1. Although it can be compiled using lower versions, our typings will begin degrading to `any`.
```
Comment (sgong-pdftron) [inline README.md:350]: avoid words such as "we" "I" in instruction

Comment (sgong-pdftron) [reply README.md:350]: it can be as simple as

- Recommended TypeScript version: 3.4.1+

File: README.md [hunk 6]
```diff
@@ -321,6 +325,35 @@ const styles = StyleSheet.create({
...
+- Due to the introduction of proper typings to PDFTron's APIs, your compiler may now give warnings or errors about your usage of these APIs. In most cases these will not prevent your app from running but it is advised that you address them to take full advantage of TypeScript. It is particularly important to address them if you use TypeScript to emit files and have enabled `noEmitOnError` in your `tsconfig.json`.
+
+### Migrating to TypeScript
+
+If you would like to migrate, [here](https://reactnative.dev/docs/typescript#adding-typescript-to-an-existing-project) is a guide on how to add TypeScript to an existing React Native project and [here](https://www.typescriptlang.org/docs/) is documentation for you to learn about the language.
```
Comment (sgong-pdftron) [inline README.md:355]: this section sounds like talking about pdftron wrapper, but it is not, im not sure whether this is needed

File: README.md (hunk 4 above)
Comment (akwanpdf) [reply README.md]: clarified

File: README.md (hunk 6 above)
Comment (akwanpdf) [reply README.md]: removed section

File: API.md (hunk 3 above)
Comment (akwanpdf) [reply API.md]: clarified; it may be helpful if they want to see full list of the exact type aliases/interfaces

**Review by sgong-pdftron — APPROVED** [review]

**Review by JamesTron — CHANGES_REQUESTED** [review]
No need to re-request review after changes are made.

**Pull request merged**